trpc/access/accessCheckers.ts

```diff
@@ -0,0 +1,36 @@
+import { Session } from 'next-auth';
+
+import { addCalclulatedGoalsFields } from '../queries/goals';
+
+import { CommentEntity, GoalEntity, ProjectEntity } from './accessEntityGetters';
+
+type AccessCheckerResult = Readonly<{ allowed: true } | { allowed: false; errorMessage: string }>;
+
+export type EntityAccessChecker<TEntity> = (session: Session, entity: TEntity) => AccessCheckerResult;
+
+const allowed = (): AccessCheckerResult => ({
+    allowed: true,
+});
+
+const notAllowed = (errorMessage: string): AccessCheckerResult => ({ allowed: false, errorMessage });
+
+export const goalAccessChecker = (session: Session, goal: GoalEntity) => {
+    const { activityId, role } = session.user;
+    const { _isEditable } = addCalclulatedGoalsFields(goal, activityId, role);
+
+    return _isEditable ? allowed() : notAllowed('No access to update Goal');
+};
+
+export const commentAccessChecker = (session: Session, comment: CommentEntity) => {
+    const { activityId } = session.user;
+
+    return comment.activityId === activityId ? allowed() : notAllowed('No access to update Comment');
+};
+
+export const projectAccessChecker = (session: Session, project: ProjectEntity) => {
+    const { activityId, role } = session.user;
+
+    return project.activityId === activityId || role === 'ADMIN'
```

Comment:
We just made _isEditable for projects, check it out https://github.com/taskany-inc/issues/issues/1629

Reply:
thanks